### PR TITLE
Add custom block for extend corends configuration

### PR DESCRIPTION
**Pull request by pipo02mix**
Towards https://github.com/giantswarm/giantswarm/issues/5389

### CODE REVIEW

File: helm/kubernetes-coredns-chart/templates/configmap.yaml [hunk 1]
```diff
@@ -10,7 +10,9 @@ metadata:
 data:
   Corefile: |
     .:{{ .Values.ports.dns.port }} {
+        {{- if not (contains "cache" .Values.configmap.custom) }}
         cache {{ .Values.configmap.cache }}
```
Comment (pipo02mix) [inline helm/kubernetes-coredns-chart/templates/configmap.yaml]: If custom block has cache then dont use default one

File: helm/kubernetes-coredns-chart/templates/configmap.yaml [hunk 2]
```diff
@@ -30,6 +32,13 @@ data:
         proxy {{ . }}
         {{- end }}
         {{- end }}
+
+        {{- if .Values.configmap.custom }}
+        {{- range (.Values.configmap.custom | trimAll "\n " |  split "\n") }}
+        {{ . }}
+        {{- end }}
+        {{- end }}
```
Comment (pipo02mix) [inline helm/kubernetes-coredns-chart/templates/configmap.yaml]: We need to render line by line or the line breaks in the text are escaped (again )and fails the conversion to json

File: helm/kubernetes-coredns-chart/values.yaml [hunk 3]
```diff
@@ -11,6 +11,7 @@ maxReplicas: 10
 
 configmap:
   cache: 30
+  custom: ""
```
Comment (pipo02mix) [inline helm/kubernetes-coredns-chart/values.yaml]: cannot be empty

**Comment by pipo02mix** [comment]
Tested on ginger

**Review by rossf7 — APPROVED** [review]
LGTM

Thanks for taking care here. For app-operator we're changing the values support from JSON to YAML so hopefully less templating magic is needed.

**Review by TheoBrigitte — APPROVED** [review]
NIce

**Pull request merged**